clang/lib/Sema/SemaType.cpp

```diff
@@ -8084,29 +8084,6 @@ static void HandleNeonVectorTypeAttr(QualType &CurType, const ParsedAttr &Attr,
         AuxTI && (AuxTI->getTriple().isAArch64() || AuxTI->getTriple().isARM());
   }
 
-  // Target must have NEON (or MVE, whose vectors are similar enough
-  // not to need a separate attribute)
-  if (!(S.Context.getTargetInfo().hasFeature("neon") ||
-        S.Context.getTargetInfo().hasFeature("mve") ||
-        S.Context.getTargetInfo().hasFeature("sve") ||
-        S.Context.getTargetInfo().hasFeature("sme") ||
-        IsTargetCUDAAndHostARM) &&
-      VecKind == VectorKind::Neon) {
-    S.Diag(Attr.getLoc(), diag::err_attribute_unsupported)
-        << Attr << "'neon', 'mve', 'sve' or 'sme'";
-    Attr.setInvalid();
-    return;
-  }
-  if (!(S.Context.getTargetInfo().hasFeature("neon") ||
```

Comment:
I believe we may need to leave still the test for MVE. We need to leave a comment in the ACLE stating that the MVE header is still needed.

Reply:
Do we ? From my understanding this checks are Neon vector types and are unrelated to MVE. The only reason MVE is used is because MVE vectors are similar enough so we can use them as neon vectors

Reply:
You can preserve the behaviour for MVE if you alter the diagnostics condition to be
`"NEON type seen" && "no MVE" && "compiling for  M-class".`